arch/x86/core/intel64/cpu.c

```diff
@@ -97,7 +97,7 @@ void arch_cpu_start(int cpu_num, k_thread_stack_t *stack, int sz,
  */
 FUNC_NORETURN void z_x86_cpu_init(struct x86_cpuboot *cpuboot)
 {
-#if defined(CONFIG_ACPI)
+#if defined(CONFIG_ACPI) && !defined(CONFIG_ACRN_COMMON)
```

Comment:
Just to clarify, why is the fix simply remove the check instead of accounting for this offset? I wonder if some other code could break as it's not expecting this "mismatch" - or is the idea exactly to find breakages and account for this difference there?

Reply:
The CPU assignment is done during ACRN configuration in the ACRN Configurator tool, and this number will not be available for Zephyr runtime to check the specific offset of the x86_cpu_loapics array. 
This change has been verified, ensuring that the code remains functional and stable.